### PR TITLE
Add charon-config.xml.j2 template.

### PR DESCRIPTION
**Pull request by hasinidilanka**
**Proposed Changes**

- Add `charon-config.xml.j2` template.

- Add the relevant keys to the `default.json` and `infer.json` files.

### CODE REVIEW

File: features/org.wso2.carbon.identity.scim2.common.feature/resources/org.wso2.carbon.identity.scim2.common.feature.infer.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "scim2.primary_authentication_scheme": {
+    "oauthbearertoken": {
```
Comment (pulasthi7) [inline features/org.wso2.carbon.identity.scim2.common.feature/resources/org.wso2.carbon.identity.scim2.common.feature.infer.json]: shall we have underscores between words

Comment (hasinidilanka) [reply features/org.wso2.carbon.identity.scim2.common.feature/resources/org.wso2.carbon.identity.scim2.common.feature.infer.json]: Fixed.

File: features/org.wso2.carbon.identity.scim2.common.feature/resources/org.wso2.carbon.identity.scim2.common.feature.default.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "scim2.enable_schema_extension": true,
+  "scim2.max_bulk_operations": "1000",
+  "scim2.max_bulk_payload": "1048576",
+  "scim2.documentation_uri": "https://docs.wso2.com/display/IS570/apidocs/SCIM2-endpoints/index.html",
```
Comment (pulasthi7) [inline features/org.wso2.carbon.identity.scim2.common.feature/resources/org.wso2.carbon.identity.scim2.common.feature.default.json]: should be 5.9.0 ?

Comment (hasinidilanka) [reply features/org.wso2.carbon.identity.scim2.common.feature/resources/org.wso2.carbon.identity.scim2.common.feature.default.json]: The latest document is for 5.7. For 5.8 and 5.9 we do not have this.

**Pull request merged**